internal/tofu/transform_diff.go

```diff
@@ -116,7 +116,7 @@ func (t *DiffTransformer) Transform(_ context.Context, g *Graph) error {
 		// A deposed instance may only have a change of Delete, Forget or NoOp.
 		// A NoOp can happen if the provider shows it no longer exists during
 		// the most recent ReadResource operation.
-		if dk != states.NotDeposed && !(rc.Action == plans.Delete || rc.Action == plans.NoOp || rc.Action == plans.Forget) {
+		if dk != states.NotDeposed && (rc.Action != plans.Delete && rc.Action != plans.NoOp && rc.Action != plans.Forget) {
```

Comment:
```suggestion
		if dk != states.NotDeposed && rc.Action != plans.Delete && rc.Action != plans.NoOp && rc.Action != plans.Forget {
```

These parens are not actually doing anything anymore in this "all ands" version, right? :thinking:

Reply:
Correct, this was the automatic fix from golang-ci-lint